plonky/src/groth16.rs

```diff
@@ -0,0 +1,113 @@
+use crate::bellman_ce::{groth16::*, Circuit};
+use crate::errors::Result;
+use crate::franklin_crypto::bellman::pairing::Engine;
+use crate::snark::SNARK;
+use rand::Rng;
+use std::marker::PhantomData;
+
+pub struct Groth16<E: Engine, C: Circuit<E>> {
+    _engine: PhantomData<E>,
+    _circuit: PhantomData<C>,
+}
+
+impl<E: Engine, C: Circuit<E>> SNARK<E::Fr> for Groth16<E, C> {
+    type Circuit = C;
+    type AssignedCircuit = C;
+    type ProvingKey = Parameters<E>;
+    type VerificationKey = VerifyingKey<E>;
+    type PreparedVerificationKey = PreparedVerifyingKey<E>;
+    type Proof = Proof<E>;
+
+    fn circuit_specific_setup<R: Rng>(
+        circuit: Self::Circuit,
+        rng: &mut R,
+    ) -> Result<(Self::ProvingKey, Self::PreparedVerificationKey)> {
+        #[cfg(debug_assertions)]
+        let t = std::time::Instant::now();
+
+        let pk = generate_random_parameters::<E, Self::Circuit, R>(circuit, rng)?;
+        let pvk = prepare_verifying_key(&pk.vk);
+
+        #[cfg(debug_assertions)]
```

Comment:
I think this timer print is not only for debugging, we also need to figure out the time cost for release.

Reply:
So remove the debug cfg here